spot-client/src/spot-tv/ui/components/meeting-frame/ApiHealthCheck.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Checks if the Jitsi API is responsive.
+ */
+export default class ApiHealthCheck {
+    /**
+     * The amount of time to wait in milliseconds for the Jitsi API to return
+     * a response.
+     */
+    static pingTimeLimit = 5000;
+
+    /**
+     * Initializes a new {@code ApiHealthCheck} instance.
+     *
+     * @param {Object} jitsiApi - An instance of JitsiMeetExternalAPI.
+     * @param {Function} onError - Callback to invoke when a health check fails.
+     */
+    constructor(jitsiApi, onError) {
+        this._jitsiApi = jitsiApi;
+        this._onError = onError;
+
+        this._destroyed = false;
+    }
+
+    /**
+     * Begins polling the Jitsi API to check if the iFrame is responsive.
+     *
+     * @returns {void}
+     */
+    start() {
+        if (this._destroyed) {
+            throw new Error('Destroyed. Create new instance.');
+        }
+
+        // Safeguard against potentially multiple checks in flight.
+        if (this._checkHealthTimeout) {
+            return;
+        }
+
+        this._checkHealthTimeout = setTimeout(() => this._checkHealth());
```

Comment:
nit: i think it's clearer to name variables like this something like: `_checkHealthTaskHandle` or something like that, since it's a handle to task, not a timeout.